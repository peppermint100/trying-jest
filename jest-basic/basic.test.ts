interface CoffeeData {
    id: number;
    name: string;
    price: number;
    size: string;
};

const data:CoffeeData[] = [
    {
        id: 1,
        name: "Americano",
        price: 4100,
        size: "Regular"
    },
    {
        id: 2,
        name: "Cafe Mocha",
        price: 5600,
        size: "Regular" 
    },
    {
        id: 3,
        name: "Mango Juice",
        price: 6300,
        size: "Large"
    }
]
test('testing test is working', () => {
    expect(3).toBe(3);
})

test('coffe data has length of 3 ', () => {
    expect(data).toHaveLength(3);
});


test('Checking Coffee\'s name', () => {
    expect(data.map(coffee => coffee.name)).toEqual([
        "Americano",
        "Cafe Mocha",
        "Mango Juice"
    ])
})


test('Testing Coffee Data Snapshot', () => {
    expect(data).toMatchSnapshot();
})