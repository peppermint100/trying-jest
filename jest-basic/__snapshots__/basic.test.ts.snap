// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Coffee Data Snapshot 1`] = `
Array [
  Object {
    "id": 1,
    "name": "Americano",
    "price": 4100,
    "size": "Regular",
  },
  Object {
    "id": 2,
    "name": "Cafe Mocha",
    "price": 5600,
    "size": "Regular",
  },
  Object {
    "id": 3,
    "name": "Mango Juice",
    "price": 6300,
    "size": "Large",
  },
]
`;
